contracts/FlowIDTableStaking.cdc

```diff
@@ -1292,6 +1401,77 @@ pub contract FlowIDTableStaking {
         return nodeIDs.keys
     }
 
+    /// Adds a node and/or a delegator to the list of node IDs who have pending token movements
+    /// or who's delegators have pending movements
+    access(contract) fun setNewMovesPending(nodeID: String, delegatorID: UInt32?) {
+        let movesPendingList = self.account.load<{String: {UInt32: Bool}}>(from: /storage/idTableMovesPendingList)
+            ?? panic("No moves pending list in account storage")
+
+        // Create an empty list of delegators with pending moves for the node ID
+        var delegatorList: {UInt32: Bool} = {}
+
+        // If there is already a list for the given node ID, overwrite the created one
+        if let existingDelegatorList = movesPendingList[nodeID] {
+            delegatorList = existingDelegatorList
+        }
+
+        // If this function call is to record a delegator's movement,
+        // record the ID
+        if let unwrappedDelegatorID = delegatorID {
+            delegatorList[unwrappedDelegatorID] = true
+        }
+
+        // Save the modified list to the node's entry
+        // If it was just a node, it will save an empty/unmodified delegator list
+        movesPendingList[nodeID] = delegatorList
+
+        self.account.save<{String: {UInt32: Bool}}>(movesPendingList, to: /storage/idTableMovesPendingList)
+    }
+
+    /// Gets a list of node IDs who have pending token movements
+    /// or who's delegators have pending movements
+    pub fun getMovesPendingList(): {String: {UInt32: Bool}}? {
+        return self.account.copy<{String: {UInt32: Bool}}>(from: /storage/idTableMovesPendingList)
+    }
+
+    /// Candidate Nodes Methods
+    /// Candidate Nodes are newly committed nodes who aren't already staked
+    /// There is a limit to the number of candidate nodes per epoch
+    /// The candidate node list is a dictionary that maps node roles
+    /// to a list of node IDs of that role
+    /// Gets the candidate node list size limit
+    pub fun getCandidateNodeLimit(): Int? {
+        return self.account.copy<Int>(from: /storage/idTableCandidateNodeLimit)
+    }
+
+    /// Adds the provided node ID to the candidate node list
+    access(contract) fun addToCandidateNodeList(nodeID: String, roleToAdd: UInt8) {
```

Comment:
This is a great catch! Definitely needs some discussing.

problem 1 is easy to solve and I can make that change. Problem 2 is a bit more difficult. In the proposal, the list was proposed to be a global list, so we'd need to discuss making it a per-node role list before implementing it. Additionally,  requiring access nodes to have a minimum stake would be a big change, because the governance team has made it clear that they do not want access nodes to have to stake, at least for now, so we are kind of in a bind with that. I'm not sure who I need to loop in though, since Paul was the main driver of this and he isn't with the company any more.

@peterargue @narangchirag Any thoughts? This is a blocker for permissionless access node